django/utils/timezone.py

```diff
@@ -20,9 +28,10 @@
     'is_aware', 'is_naive', 'make_aware', 'make_naive',
 ]
 
-
+# ???: Can we have a fallback for this to pytz.utc?
 # UTC time zone as a tzinfo instance.
-utc = pytz.utc
+utc = zoneinfo.ZoneInfo('UTC')
```

Comment:
Maybe I'm missing something, but do we need to use `zoneinfo.ZoneInfo('UTC')` or can we just use `timezone.utc` (i.e. `datetime.timezone.utc`)?

If we want a fallback, I guess we could scrub this and use module-level `__getattr__`:

```python
def __getattr__(name):
    if name == 'utc':
        return pytz.utc if settings.USE_PYTZ_DEPRECATION_SHIM else timezone.utc
    raise AttributeError(f'module {__name__!r} has no attribute {name!r}')
```

We could possibly even adjust that to raise a deprecation warning if necessary:

```python
def __getattr__(name):
    if name != 'utc':
        raise AttributeError(f'module {__name__!r} has no attribute {name!r}')
    if not settings.USE_PYTZ_DEPRECATION_SHIM:
        return timezone.utc  # XXX: Or zoneinfo.ZoneInfo('UTC')
    warnings.warn(..., RemovedInDjango50Warning, stacklevel=3)
    return pytz.utc
```

Reply:
> ...can we just use timezone.utc (i.e. datetime.timezone.utc)?

I guess that's where I'm unsure… — that too is a change, but would it be OK? 🤔

> ...  use module-level `__getattr__`...

Let me have a play with that. It's ≈what we want, but I didn't want to hit `settings` at the module level so this approach might be right.

Reply:
Is there a reason preventing us from doing this?

```python
if settings.USE_PYTZ_DEPRECATION_SHIM:
    import pytz
    utc = pytz.utc
else:
    utc = zoneinfo.ZoneInfo('UTC')
```

Reply:
Hey @aaugustin — thanks for the comments! 

> Is there a reason preventing us from doing this?

Just the comment on `get_default_timezone()`...

>     # In order to avoid accessing settings at compile time,
>     # wrap the logic in a function and cache the result.

Assuming that would apply here too. 

I'll give Nick's suggestion a go.